src/commands/publish.rs

```diff
@@ -11,19 +12,34 @@ use crate::settings::global_user::GlobalUser;
 use crate::settings::toml::{DeployConfig, Target};
 use crate::sites;
 use crate::terminal::emoji;
-use crate::terminal::message::{Message, StdOut};
+use crate::terminal::message::{Message, StdErr, StdOut};
 use crate::upload;
 
+#[derive(Serialize, Deserialize, Default)]
+pub struct PublishOutput {
+    pub success: bool,
+    pub name: String,
+    pub urls: Vec<String>,
+}
+
 pub fn publish(
     user: &GlobalUser,
     target: &mut Target,
     deploy_config: DeployConfig,
+    is_json_out: bool,
 ) -> Result<(), failure::Error> {
     validate_target_required_fields_present(target)?;
 
-    // Build the script before uploading.
-    build_target(&target)?;
-
+    // Build the script before uploading and log build result
+    let build_result = build_target(&target);
+    match build_result {
+        Ok(msg) => {
+            StdErr::success(&msg);
+            Ok(())
+        }
+        Err(e) => Err(e),
+    }?;
+    let mut jsonout = PublishOutput::default();
```

Comment:
we'll always only instantiate this once per branch, so rather than doing this out here, just construct the struct at the point of success.